### ISSUE TITLE
`\N{name}` needs to use fuzzy matching

### ISSUE THREAD

**Issue by natecook1000** [apple/swift-experimental-string-processing#310]
The name in a `\N{...}` block should use fuzzy matching instead of exact equality.

```swift
"👩‍👩‍👧‍👦".contains(/(?u).\N{ZERO WIDTH JOINER}/)   // true
"👩‍👩‍👧‍👦".contains(/(?u).\N{ZeroWidthJoiner}/)     // false, should be true
```

**Comment by milseman** [comment]
Status?

**Comment by natecook1000** [comment]
This was resolved in #382.
